### PR TITLE
feat: support aten.clamp.Tensor and update aten.clamp.default dynamo converters

### LINKED ISSUE #2517: `aten.clamp.Tensor`

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

support `aten.clamp.Tensor` and update `aten.clamp.default` dynamo converters

Fixes #2517 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py [hunk 1]
```diff
@@ -186,63 +184,22 @@ def clamp(
...
-    if max_val is not None:
-        clamp_max_layer = _add_layer(
-            ctx, input_val, max_val, trt.ElementWiseOperation.MIN, name
-        )
-        set_layer_name(clamp_max_layer, target, f"{name}_clamp_max")
-        input_val = clamp_max_layer.get_output(0)
+    if min_val is None:
+        min_val = float("-inf")
+    if max_val is None:
+        max_val = float("inf")
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py:193]: In the case that one of these is not specified, can `convert_binary_elementwise` + TRT handle infinity as an argument? Would it be possible to instead omit that `min` or `max` layer from the computation entirely? As in:
```python
clamp_min = input_val
if min_val is not None:
    # Compute clamp min
    clamp_min = ...

clamp_max = clamp_min
if max_val is not None:
    # Compute clamp max
    clamp_max = ...

return clamp_max

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py]: That makes sense! Thanks for pointing out!

**Review by gs-olive — COMMENTED** [review]
Overall looks good - it seems that a few of the `test_clip` converter tests are failing, however

**Review by gs-olive — APPROVED** [review]
Looks good to me!

**Pull request merged**